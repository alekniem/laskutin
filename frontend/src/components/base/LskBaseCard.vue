<template>
    <v-card v-bind="$attrs">
        <slot name="header" />
        <v-divider />
        <slot />
    </v-card>
</template>

<script>
export default {
    name: "LskBaseCard",
};
</script>
