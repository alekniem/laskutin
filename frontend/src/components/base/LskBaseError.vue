<template>
    <div class="red--text font-weight-bold">
        <slot />
    </div>
</template>

<script>
export default {
    name: "LskBaseError",
};
</script>
