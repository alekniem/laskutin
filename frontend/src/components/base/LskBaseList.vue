<template>
    <v-list :id="cid" dense>
        <slot />
    </v-list>
</template>

<script>
export default {
    name: "LskBaseList",

    props: {
        cid: { type: String, required: true },
    },
};
</script>
