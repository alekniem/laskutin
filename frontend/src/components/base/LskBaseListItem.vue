<template>
    <v-list-item :id="listItemId">
        <v-list-item-content
            class="text-caption blue--text text--darken-2 font-weight-bold"
        >
            {{ text | trans }}
        </v-list-item-content>
        <v-list-item-content>
            {{ value }}
        </v-list-item-content>
    </v-list-item>
</template>

<script>
import camelCase from "lodash/camelCase";
import upperFirst from "lodash/upperFirst";

export default {
    name: "LskBaseListItem",

    props: {
        itemId: { type: String, required: true },
        parentId: { type: String, required: true },
        text: { type: String, required: true },
        value: { type: [String, Number], required: true },
    },

    computed: {
        listItemId() {
            return (
                this.parentId + upperFirst(camelCase(this.itemId)) + "ListItem"
            );
        },
    },
};
</script>
