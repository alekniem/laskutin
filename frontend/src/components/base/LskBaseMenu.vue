<template>
    <v-menu bottom left>
        <template #activator="{ attrs, on }">
            <v-btn :id="activatorId" icon v-bind="attrs" v-on="on">
                <v-icon>mdi-dots-vertical</v-icon>
            </v-btn>
        </template>
        <v-list>
            <slot />
        </v-list>
    </v-menu>
</template>

<script>
export default {
    name: "LskBaseMenu",

    props: {
        activatorId: { type: String, required: true },
    },
};
</script>
