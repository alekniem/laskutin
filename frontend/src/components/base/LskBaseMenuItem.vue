<template>
    <v-list-item :id="cid" link>
        <v-list-item-title>
            {{ text | trans }}
        </v-list-item-title>
    </v-list-item>
</template>

<script>
export default {
    name: "LskBaseMenuItem",

    props: {
        cid: { type: String, required: true },
        text: { type: String, required: true },
    },
};
</script>
