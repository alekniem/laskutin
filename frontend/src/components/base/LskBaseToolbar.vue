<template>
    <v-toolbar flat>
        <v-toolbar-title :id="titleId">
            <slot />
        </v-toolbar-title>
        <v-spacer />
        <slot name="button" />
    </v-toolbar>
</template>

<script>
export default {
    name: "LskBaseToolbar",

    props: {
        titleId: { type: String, required: true },
    },
};
</script>
