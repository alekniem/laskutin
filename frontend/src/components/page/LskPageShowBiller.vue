<template>
    <div>
        <v-row v-if="billerData">
            <v-col cols="12" xs="12" sm="8" md="7" lg="4">
                <LskBaseCard>
                    <template #header>
                        <LskBaseToolbar :titleId="cid + 'Title'">
                            <template #button>
                                <LskBaseMenu :activatorId="cid + 'MenuActivator'">
                                    <LskBaseMenuItem
                                        @click.native="editBiller"
                                        :cid="cid + 'EditBillerMenuItem'"
                                        text="Edit biller"
                                    />
                                </LskBaseMenu>
                            </template>
                            {{ "Biller information" | trans }}
                        </LskBaseToolbar>
                    </template>
                    <LskBaseList :cid="cid + 'List'">
                        <LskBaseListItem
                            v-for="(itemData, itemId) in billerData"
                            :itemId="itemId"
                            :key="itemId"
                            :parentId="cid"
                            :text="itemData.text"
                            :value="itemData.value"
                        />
                    </LskBaseList>
                </LskBaseCard>
            </v-col>
        </v-row>

        <LskBaseError v-if="errorMessage">
            {{ errorMessage }}
        </LskBaseError>

        <LskDialogAddOrEditBiller
            :cid="cid + 'EditBillerDialog'"
            :currentData="rawData"
            :id="id"
            :key="keyEditBiller"
            v-model="showEditBiller"
        />
    </div>
</template>

<script>
import lsk_api from "@/plugins/lsk_api";

export default {
    name: "LskPageShowBiller",

    props: {
        id: { type: [String, Number], required: true },
    },

    data() {
        return {
            apiActionName: "billerShow",
            cid: "billerShowPage",
            billerData: null,
            errorMessage: null,
            rawData: {},

            counterEditBiller: 0,
            showEditBiller: false,
        };
    },

    computed: {
        keyEditBiller() {
            return "edit_biller_" + this.counterEditBiller;
        },
    },

    watch: {
        showEditBiller(value) {
            if (!value) this.queryApi();
        },
    },

    mounted() {
        this.queryApi();
    },

    methods: {
        editBiller() {
            this.counterEditBiller++;
            this.showEditBiller = true;
        },

        queryApi() {
            lsk_api
                .query(this.apiActionName, this.id, null)
                .then((response) => this.parseResponse(response.data))
                .catch((error) => this.parseError(error.response.data));
        },

        parseResponse(data) {
            this.rawData = data;

            this.billerData = {
                id: { text: "Biller id", value: data.id },
                name: { text: "Name", value: data.name },
                address_line_one: {
                    text: "Address line 1",
                    value: data.address_line_one,
                },
                address_line_two: {
                    text: "Address line 2",
                    value: data.address_line_two,
                },
                email: { text: "Email", value: data.email },
                phone_number: {
                    text: "Phone number",
                    value: data.phone_number,
                },
                business_identity_code: {
                    text: "Business identity code",
                    value: data.business_identity_code,
                },
                bank_name: { text: "Bank name", value: data.bank_name },
                bank_iban: { text: "Bank iban", value: data.bank_iban },
                bank_bic: { text: "Bank bic", value: data.bank_bic },
                pdf_title: { text: "Pdf title", value: data.pdf_title },
                pdf_author: { text: "Pdf author", value: data.pdf_author },
                created_at: {
                    text: "Created",
                    value: this.formatDate(
                        data.created_at,
                        "D.M.YYYY HH:mm:ss"
                    ),
                },
                updated_at: {
                    text: "Updated",
                    value: this.formatDate(
                        data.updated_at,
                        "D.M.YYYY HH:mm:ss"
                    ),
                },
            };

            this.errorMessage = null;
        },

        parseError(data) {
            this.rawData = {};
            this.billerData = null;
            this.errorMessage = data.message || "Failed to get data";
        },
    },
};
</script>
