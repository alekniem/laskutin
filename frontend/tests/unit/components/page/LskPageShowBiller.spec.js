import axios from "axios";
import flushPromises from "flush-promises";
import LskBaseCard from "@/components/base/LskBaseCard";
import LskBaseError from "@/components/base/LskBaseError";
import LskBaseList from "@/components/base/LskBaseList";
import LskBaseListItem from "@/components/base/LskBaseListItem";
import LskBaseMenu from "@/components/base/LskBaseMenu";
import LskBaseMenuItem from "@/components/base/LskBaseMenuItem";
import LskBaseToolbar from "@/components/base/LskBaseToolbar";
import LskPageShowBiller from "@/components/page/LskPageShowBiller";
import { createWrapper } from "UT/lsk_test_helper";

jest.mock("axios");

axios.get.mockImplementation((url) => {
    if (url === "biller/123") {
        return Promise.resolve({
            data: {
                id: 123,
                name: "NAME",
                address_line_one: "ADDRESS LINE ONE",
                address_line_two: "ADDRESS LINE TWO",
                email: "EMAIL",
                phone_number: "PHONE NUMBER",
                business_identity_code: "BUSINESS IDENTITY CODE",
                bank_name: "BANK NAME",
                bank_iban: "BANK IBAN",
                bank_bic: "BANK BIC",
                pdf_title: "PDF TITLE",
                pdf_author: "PDF AUTHOR",
                created_at: "2021-01-01 12:00:01",
                updated_at: "2021-02-02 12:00:02",
            },
        });
    }
    return Promise.reject({
        response: {
            data: {
                message: "GET QUERY ERROR",
            },
        },
    });
});

const stubs = {
    LskBaseCard,
    LskBaseError,
    LskBaseList,
    LskBaseListItem,
    LskBaseMenu,
    LskBaseMenuItem,
    LskBaseToolbar,
    LskDialogAddOrEditBiller: true,
};

describe("LskPageShowBiller", () => {
    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskPageShowBiller, {
            propsData: {
                id: 123,
            },
            stubs,
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("page");
    });

    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskPageShowBiller, {
            propsData: {
                id: "INVALID",
            },
            stubs,
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("error");
    });

    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskPageShowBiller, {
            propsData: {
                id: 123,
            },
            stubs,
        });
        await flushPromises();

        const menu = wrapper.findComponent(LskBaseMenu);
        menu.find("#billerShowPageMenuActivator").trigger("click");
        await flushPromises();
        expect(menu.html()).toMatchSnapshot("menu");

        menu.find("#billerShowPageEditBillerMenuItem").trigger("click");
        await flushPromises();
        const dialog = wrapper.find("LskDialogAddOrEditBiller-stub");
        expect(dialog.html()).toMatchSnapshot("dialog edit biller");
    });
});
