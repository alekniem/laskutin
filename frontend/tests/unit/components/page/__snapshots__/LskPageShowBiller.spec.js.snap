// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskPageShowBiller should match snapshot: dialog edit biller 1`] = `
<lskdialogaddoreditbiller-stub cid="billerShowPageEditBillerDialog"
                               currentdata="[object Object]"
                               id="123"
                               value="true"
>
</lskdialogaddoreditbiller-stub>
`;

exports[`LskPageShowBiller should match snapshot: error 1`] = `
<div>
  <div class="red--text font-weight-bold">
    GET QUERY ERROR
  </div>
  <lskdialogaddoreditbiller-stub cid="billerShowPageEditBillerDialog"
                                 currentdata="[object Object]"
                                 id="INVALID"
  >
  </lskdialogaddoreditbiller-stub>
</div>
`;

exports[`LskPageShowBiller should match snapshot: menu 1`] = `
<div class="v-menu">
  <transition-stub name="v-menu-transition">
    <div role="menu"
         class="v-menu__content theme--light menuable__content__active "
         style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 8; display: none;"
    >
      <div class="v-list v-sheet theme--light">
        <div tabindex="0"
             id="billerShowPageEditBillerMenuItem"
             role="menuitem"
             class="v-list-item v-list-item--link theme--light"
        >
          <div class="v-list-item__title">
            Edit biller
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`LskPageShowBiller should match snapshot: page 1`] = `
<div>
  <div class="row">
    <div xs="12"
         class="col-sm-8 col-md-7 col-lg-4 col-12"
    >
      <div class="v-card v-sheet theme--light">
        <header class="v-sheet theme--light v-toolbar v-toolbar--flat"
                style="height: 64px;"
        >
          <div class="v-toolbar__content"
               style="height: 64px;"
          >
            <div id="billerShowPageTitle"
                 class="v-toolbar__title"
            >
              Biller information
            </div>
            <div class="spacer">
            </div>
            <div class="v-menu">
            </div>
            <button type="button"
                    class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                    id="billerShowPageMenuActivator"
                    role="button"
                    aria-haspopup="true"
                    aria-expanded="false"
            >
              <span class="v-btn__content">
                <i aria-hidden="true"
                   class="v-icon notranslate mdi mdi-dots-vertical theme--light"
                >
                </i>
              </span>
            </button>
          </div>
        </header>
        <hr role="separator"
            aria-orientation="horizontal"
            class="v-divider theme--light"
        >
        <div role="list"
             id="billerShowPageList"
             class="v-list v-sheet theme--light v-list--dense"
        >
          <div tabindex="-1"
               id="billerShowPageIdListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Biller id
            </div>
            <div class="v-list-item__content">
              123
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageNameListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Name
            </div>
            <div class="v-list-item__content">
              NAME
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageAddressLineOneListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Address line 1
            </div>
            <div class="v-list-item__content">
              ADDRESS LINE ONE
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageAddressLineTwoListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Address line 2
            </div>
            <div class="v-list-item__content">
              ADDRESS LINE TWO
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageEmailListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Email
            </div>
            <div class="v-list-item__content">
              EMAIL
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPagePhoneNumberListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Phone number
            </div>
            <div class="v-list-item__content">
              PHONE NUMBER
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageBusinessIdentityCodeListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Business identity code
            </div>
            <div class="v-list-item__content">
              BUSINESS IDENTITY CODE
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageBankNameListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Bank name
            </div>
            <div class="v-list-item__content">
              BANK NAME
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageBankIbanListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Bank iban
            </div>
            <div class="v-list-item__content">
              BANK IBAN
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageBankBicListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Bank bic
            </div>
            <div class="v-list-item__content">
              BANK BIC
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPagePdfTitleListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Pdf title
            </div>
            <div class="v-list-item__content">
              PDF TITLE
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPagePdfAuthorListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Pdf author
            </div>
            <div class="v-list-item__content">
              PDF AUTHOR
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageCreatedAtListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Created
            </div>
            <div class="v-list-item__content">
              1.1.2021 12:00:01
            </div>
          </div>
          <div tabindex="-1"
               id="billerShowPageUpdatedAtListItem"
               role="listitem"
               class="v-list-item theme--light"
          >
            <div class="v-list-item__content text-caption blue--text text--darken-2 font-weight-bold">
              Updated
            </div>
            <div class="v-list-item__content">
              2.2.2021 12:00:02
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <lskdialogaddoreditbiller-stub cid="billerShowPageEditBillerDialog"
                                 currentdata="[object Object]"
                                 id="123"
  >
  </lskdialogaddoreditbiller-stub>
</div>
`;
